src/identity/identity-wallet.ts

```diff
@@ -661,13 +661,28 @@ export class IdentityWallet implements IIdentityWallet {
       allowedIssuers: [did.string()]
     });
 
-    if (credentials.length) {
+    if (credentials.length > 1) {
+      throw new Error('more than 1 auth credential for the same key is located in storage');
```

Comment:
Do we really need this check? It may block users which for some reason already have 2 credentials with the same key. We can return correct credential (if it exists - great, if not - create) and remove all the other credentials for the same key.

Reply:
fixed